components/mosquitto/README.md

```diff
@@ -0,0 +1,32 @@
+# ESP32 Mosquitto Port
+
+This is a lightweight port of the Mosquitto broker designed to run on the ESP32. It currently supports a single listener and TCP transport only.
+
+## Supported Options
+
+The Espressif port supports a limited set of options (with plans to add more in future releases). These options can be configured through a structure passed to the `mosq_broker_start()` function. For detailed information on available configuration options, refer to the [API documentation](api.md).
+
+## API
+
+### Starting the Broker
+
+To start the broker, call the `mosq_broker_start()` function with a properly configured settings structure. The broker operates in the context of the calling task and does not create a separate task.
```

Comment:
Why not create a separate task to run? This would make the usage inline with other components we have.

Reply:
This can be added later, if needs be.

I'd like to include this "task-less" option, because it's simply more flexible. Otherwise we'd start the task in the lib, but had to expose 
* task memory allocation capabilities
* task affinity
* priority / stack size
* ...something else?

Reply:
I'll address this in a followup PR:
* rename this API to `mqtt_broker_run()` to run the broker within current task
* use `mosq_broker_start()` `mosq_broker_stop()` to start/stop a task with the broker.